lib/child_process.js

```diff
@@ -111,28 +111,27 @@ const MAX_BUFFER = 1024 * 1024;
  *   }} [options]
  * @returns {ChildProcess}
  */
-function fork(modulePath /* , args, options */) {
+function fork(modulePath, args = [], options = {}) {
   modulePath = getValidatedPath(modulePath, 'modulePath');
 
   // Get options and args arguments.
   let execArgv;
-  let options = {};
-  let args = [];
-  let pos = 1;
-  if (pos < arguments.length && ArrayIsArray(arguments[pos])) {
-    args = arguments[pos++];
-  }
 
-  if (pos < arguments.length && arguments[pos] == null) {
-    pos++;
+  if (args == null) {
+    args = [];
+  } else if (typeof args !== 'object') {
+    throw new ERR_INVALID_ARG_VALUE('args', args);
+  } else if (!ArrayIsArray(args)) {
+    options = args;
+    args = [];
   }
 
-  if (pos < arguments.length && arguments[pos] != null) {
-    if (typeof arguments[pos] !== 'object') {
-      throw new ERR_INVALID_ARG_VALUE(`arguments[${pos}]`, arguments[pos]);
-    }
-
-    options = { ...arguments[pos++] };
+  if (options == null) {
+    options = {};
+  } else if (typeof options !== 'object') {
+    throw new ERR_INVALID_ARG_VALUE('options', options);
+  } else {
+    options = { ...options };
   }
```

Comment:
I think this should be `ERR_INVALID_ARG_TYPE`. We can use a validator to help with this:
```suggestion
  validateObject(options, 'options', { nullable: true });
  options = { ...options };
```

Reply:
This makes it so that `options` cannot be omitted or undefined. Is that intentional?

Reply:
Or was something closer to this intended?

```suggestion
  if (options == undefined) {
    options = {};
  }
  validateObject(options, 'options', { nullable: true });
  options = { ...options };
```

Reply:
```js
if (options == undefined) {
    options = {};
}
options = { ...options };
```
is equivalent to 
```js
options = { ...options };
```
Or am I missing something?

Reply:
I'll go with my modification above (but in a separate semver-major PR).

Reply:
> Or am I missing something?

I think you're overlooking that `validateObject(options, 'options', { nullable: true });` will throw when `options` is undefined.

Reply:
I guess this would work?

```
if (options != null) {
  validateObject(options, 'options');
}
options = { ...options };
```

Reply:
> I think you're overlooking that `validateObject(options, 'options', { nullable: true });` will throw when `options` is `undefined`.

Yep, that's exactly that 😅 With that in mind, I agree that https://github.com/nodejs/node/pull/41280#discussion_r774747281 is the ideal choice (with the current edit).

Reply:
https://github.com/nodejs/node/pull/41305